packages/asset-swapper/src/network/samplers/utils/pools_cache.ts

```diff
@@ -45,6 +59,9 @@ export abstract class PoolsCache {
             return undefined;
         }
         if (PoolsCache._isExpired(value)) {
+            // Auto-refresh the pools if expired.
+            this.getFreshPoolsForPairAsync(takerToken, makerToken)
+                .catch(err => console.error(err));
```

Comment:
Previously we would manually reach into the sampler and refresh the pools for the polled tokens in `MarketOperationUtils` (yuck). Now we just auto-refresh every time pairs are touched.